src/main/java/core/basesyntax/service/operation/BalanceOperationHandler.java

```diff
@@ -0,0 +1,14 @@
+package core.basesyntax.service.operation;
+
+import core.basesyntax.service.FruitTransaction;
+import java.util.Map;
+
+public class BalanceOperationHandler implements OperationHandler {
+    @Override
+    public void handle(FruitTransaction transaction, Map<String, Integer> storage) {
+        String fruit = transaction.getFruit();
+        int amount = transaction.getAmount();
+
+        storage.put(fruit, storage.getOrDefault(fruit, 0) + amount);
```

Comment:
According to the task description, the `Balance` operation should set a new balance for a fruit, not add to the existing one. So, this line should be `storage.put(fruit, amount);`